### PR TITLE
Migrate test utils to vitest

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Migrate test utils to vitest.

# Exception

- `html.utils.spec` -> I'll take care of migrating that one, I'll change the way we load marked library (done in #3524)
- `utils.spec` -> can you take care of that one @dskloetd ?

### CODE REVIEW

**Comment by dskloetd** [comment]
> * `utils.spec` -> can you take care of that one @dskloetd ?

I'm not sure what I'm being asked here?

**Comment by dskloetd** [comment]
Is something missing? I'm not sure what in particular you mean by `utils.spec`.
And can you give some context on what you had trouble with?

File: frontend/vitest.config.ts [hunk 1]
```diff
@@ -44,6 +44,11 @@ export default defineConfig(
     },
     test: {
       environment: "jsdom",
+      environmentOptions: {
+        jsdom: {
+          url: "https://nns.internetcomputer.org/",
```
Comment (dskloetd) [inline frontend/vitest.config.ts:49]: What is this needed for?

Comment (peterpeterparker) [reply frontend/vitest.config.ts:49]: It replicates the configuration of Jest. It is useful as some test relies on the window url context, notably the sns aggregator utils spec test of this PR.

**Comment by peterpeterparker** [comment]
> And can you give some context on what you had trouble with?

Yes sure. Let me open the draft and migrate most of things.

**Review by dskloetd — APPROVED** [review]

**Comment by peterpeterparker** [comment]
Opened a draft #3529

**Pull request merged**